File: consumer/src/test/java/ru/romanow/protocols/consumer/service/PingServiceTest.java
package ru.romanow.protocols.consumer.service;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.cloud.contract.stubrunner.spring.AutoConfigureStubRunner;
import org.springframework.cloud.contract.stubrunner.spring.StubRunnerProperties;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;

import static junit.framework.TestCase.assertTrue;

@ActiveProfiles("test")
@SpringBootTest
@RunWith(SpringRunner.class)
@AutoConfigureStubRunner(
        ids = "ru.romanow.protocols:producer:+:stubs:8080",
        stubsMode = StubRunnerProperties.StubsMode.LOCAL)
public class PingServiceTest {

    @Autowired
    private PingService pingService;

    @Test
    public void testMakeOperation() {
        boolean available = pingService.ping();
        assertTrue(available);
    }
}

File: consumer/src/test/java/ru/romanow/protocols/consumer/service/PingServiceTest.kt
package ru.romanow.protocols.consumer.service

import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.cloud.contract.stubrunner.spring.AutoConfigureStubRunner
import org.springframework.cloud.contract.stubrunner.spring.StubRunnerProperties
import org.springframework.test.context.ActiveProfiles

@ActiveProfiles("test")
@SpringBootTest
@AutoConfigureStubRunner(
    ids = ["ru.romanow.protocols:producer:+:stubs:8080"],
    stubsMode = StubRunnerProperties.StubsMode.LOCAL
)
internal class PingServiceTest {

    @Autowired
    private lateinit var pingService: PingService

    @Test
    fun testMakeOperation() {
        val available = pingService.ping()
        Assertions.assertThat(available).isTrue
    }
}